tests/solidity/test_close_short.py

```diff
@@ -0,0 +1,364 @@
+"""Close long market trade tests that match those being executed in the solidity repo"""
+import unittest
+
+import elfpy.agents.agent as agent
+import elfpy.markets.hyperdrive.hyperdrive_market as hyperdrive_market
+import elfpy.pricing_models.hyperdrive as hyperdrive_pm
+import elfpy.time as time
+import elfpy.types as types
+
+
+# pylint: disable=too-many-arguments
+# pylint: disable=duplicate-code
+
+
+class TestCloseShort(unittest.TestCase):
+    """
+    Test opening a opening and closing a short in hyperdrive
+    3 failure cases: zero amount, invalid amount, invalid timestamp
+    6 success cases:
+        - close immediately, with regular, and small amounts
+        - redeem at maturity, with zero interest, and with negative interest (skipped)
+        - close halfway thru term, with zero, and negative interest (both skipped)
+    """
+
+    contribution: float = 500_000_000
+    target_apr: float = 0.05
+    term_length: int = 365
+    alice: agent.Agent
+    bob: agent.Agent
+    celine: agent.Agent
+    hyperdrive: hyperdrive_market.Market
+
+    def setUp(self):
+        """Set up agent, pricing model, & market for the subsequent tests.
+        This function is run before each test method.
+        """
+        self.alice = agent.Agent(wallet_address=0, budget=self.contribution)
+        self.bob = agent.Agent(wallet_address=1, budget=self.contribution)
+        block_time = time.BlockTime()
+        pricing_model = hyperdrive_pm.HyperdrivePricingModel()
+        market_state = hyperdrive_market.MarketState(
+            trade_fee_percent=0.0,
+            redemption_fee_percent=0.0,
+        )
+        self.hyperdrive = hyperdrive_market.Market(
+            pricing_model=pricing_model,
+            market_state=market_state,
+            position_duration=time.StretchedTime(
+                days=365, time_stretch=pricing_model.calc_time_stretch(self.target_apr), normalizing_constant=365
```

Comment:
I wanted to keep this as position_duration because it stores a StretchedTime object, as is the case elsewhere in the codebase.

I chose term_length, although synonymous with position duration, to differentiate a variable that stores a scalar instead.

Reply:
I meant change '365' to term length

Reply:
oh lol, yes I did that!